PETWorks/arx.py

```diff
@@ -50,20 +51,33 @@ def loadDataHierarchy(
 
 
 def setDataHierarchies(
-    data: Data, hierarchies: dict[str, list[list[str]]], attributeTypes: dict
+    data: Data,
+    hierarchies: Dict[str, Hierarchy],
+    attributeTypes: Dict[str, str],
 ) -> None:
-    for attributeName, hierarchy in hierarchies.items():
-        data.getDefinition().setAttributeType(attributeName, hierarchy)
-        attributeType = attributeTypes.get(attributeName)
-
+    for attributeName, attributeType in attributeTypes.items():
+        if attributeName in hierarchies.keys():
+            if attributeType == QUASI_IDENTIFIER:
+                data.getDefinition().setAttributeType(
+                    attributeName, hierarchies[attributeName]
+                )
```

Comment:
I recommend adding `continue` at this line and removing lines 69~70, and 78. This way, the code is smaller.